src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurfaceBase.cs

```diff
@@ -133,9 +122,17 @@ private void UnoGLDrawingArea_Render(object o, RenderArgs args)
 			// update the control
 			canvas.Flush();
 
-			_gl.Flush();
+			GLFlush();
 		}
 
+		protected abstract void GLClear();
+
+		protected abstract void GLFlush();
+
+		protected abstract (int framebuffer, int stencil, int samples)GetGLBuffers();
```

Comment:
It seems that all 3 methods implementations are identical* for both GL and GLES subclasses. 

Considering virtual calls are often (if not devirtualized) slower than normal calls I think it might be best to have the base class make the calls directly. 

`*` If the situation ever change (any implementation become different) then the override(s) can be introduced at that time.

Reply:
Those are unfortunately not identical, as the namespace is different. I agree that the virtual call will be impacting a bit here. We could do the switching in-class, at the expense of a bit of readability.